astrapy/db.py

```diff
@@ -1753,6 +1761,14 @@ async def delete_subdocument(self, id: str, subdoc: str) -> API_RESPONSE:
         return response
 
     async def upsert(self, document: API_DOC) -> str:
+        # Deprecated: call the upsert_one method
+        DEPRECATION_MESSAGE = "Method 'upsert' of AstraDBCollection is deprecated. Please switch to method 'upsert_one'."
+
+        warn(DEPRECATION_MESSAGE, DeprecationWarning, stacklevel=2)
```

Comment:
We could factorize in a `deprecation_warning` method ?

Reply:
Agree- maybe a method decorator is even less intrusive. For that matter we might just use [this](https://pypi.org/project/deprecation/).

Reply:
I like this idea! I'm going to push an update now which uses the deprecation module. it looks very dev friendly, and seems to work with our tests. We can easily update the messages etc and now they're just decorators on the function itself. See the update and let me know if you guys have any further feedback @cbornet @hemidactylus . Thanks :)